include/core/rheological_model.h

```diff
@@ -0,0 +1,114 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2019 -  by the Lethe authors
+ *
+ * This file is part of the Lethe library
+ *
+ * The Lethe library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+#ifndef lethe_rheological_model_h
+#define lethe_rheological_model_h
+
+#include <core/parameters.h>
+
+using namespace dealii;
+
+/**
+ * @brief RheologicalModel. Abstract class that allows to calculate the
+ * non-newtonian viscosity on each quadrature point and the shear rate
+ * magnitude. RheologicalModel::get_viscosity() is a pure virtual method,
+ * since it can only be calculated knowing the rheological model that's
+ * being used.
+ */
+template <int dim>
+class RheologicalModel
+{
+public:
+  /**
+   * @brief Default constructor
+   */
+  RheologicalModel()
+  {}
+
+  /**
+   * @brief Returns the non-newtonian viscosity.
+   *
+   * @param shear_rate The shear rate tensor at the position of the
+   * considered quadratured point
+   */
+  virtual double
+  get_viscosity(const Tensor<2, dim> shear_rate) = 0;
```

Comment:
It would be better to pass a reference instead of passing a copy. This will prevent unnecessary copying of the tensor for reasons which are not necessary.
```suggestion
  get_viscosity(const Tensor<2, dim> & shear_rate) = 0;
```